app/javascript/greetings.js

```diff
@@ -0,0 +1,20 @@
+import React, { useEffect, useState } from 'react';
+
+const Greeting = () => {
+  const [greeting, setGreeting] = useState('');
+
+  useEffect(() => {
+    fetch('/api/messages/random_greeting')
+      .then(response => response.json())
+      .then(data => setGreeting(data.greeting))
+      .catch(error => console.error(error));
+  }, []);
+
```

Comment:
- [ ] In your greeting component, kindly remove the api request here and instead call the action of your greeting reducer which handles the api request.